GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -801,20 +801,84 @@ protected override void DisposeCustomResources()
                 _isDetachedHead = DetachedHeadParser.IsDetachedHead(currentBranch);
+                string[] regexes = AppSettings.PrioritizedBranchNames.Split(";", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+                string[] localBranchRegexes = regexes.Select(regex => $"^({regex})$").ToArray();
+                string[] remoteBranchRegexes = regexes.Select(regex => $"^{_remoteBranchPrefix}[^/]+/({regex})$").ToArray();
+                string[] remoteRegexes = AppSettings.PrioritizedRemoteNames.Split(";", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)
+                    .Select(regex => $"^{_remoteBranchPrefix}({regex})/").ToArray();
+
+                foreach (string branch in branches)
+                {
+                    _orderByBranch[branch] = GetBranchOrder(branch);
+                }
+
+                return;
+
+                // Get the order for each branch.
+                // Add max possible order value to next "level" to sort properly with the order for each regex.
+                int GetBranchOrder(string branch)
+                {
+                    return branch == _currentBranch ? 0
+                        : 1
+                        + (
+                            TryGetOrder(branch, localBranchRegexes, out int localBranchOrder)
+
+                            // Matching local branch
+                            ? localBranchOrder
+                            : localBranchRegexes.Length
+                            + (
+                                TryGetOrder(branch, remoteBranchRegexes, out int remoteBranchOrder)
+                                ? (
+                                    TryGetOrder(branch, remoteRegexes, out int remoteOrder)
+
+                                    // prioritized branch, order per branch prio then remote
+                                    ? ((remoteBranchOrder * remoteRegexes.Length) + remoteOrder)
+
+                                    // prioritized branch but not remote, order after all prioritized remotes
+                                    : ((remoteBranchRegexes.Length * remoteRegexes.Length) + remoteBranchOrder))
+                                : (remoteBranchRegexes.Length * (remoteRegexes.Length + 1))
+                                + (
+                                    IsLocalBranch()
+
+                                    // Not prioritized local branch
+                                    ? 0
+                                    : TryGetOrder(branch, remoteRegexes, out remoteOrder)
+
+                                    // Not prioritized branch on prioritized remote (note that remoteOrder was not evaluated previously)
+                                    ? 1 + remoteOrder
+
+                                    // Non match, return int.Max
+                                    : int.MaxValue - 1 - localBranchRegexes.Length - (remoteBranchRegexes.Length * (1 + remoteRegexes.Length)))));
```

Comment:
Calculation to `int.MaxValue` is not necessary. More consistent:
```suggestion
                                    : 1
                                    + (
                                          TryGetOrder(branch, remoteRegexes, out remoteOrder)

                                          // Not prioritized branch on prioritized remote
                                          ? remoteOrder
                                          : remoteRegexes.Length))));
```

Reply:
Used a variant of this, 

Also changing the sort rules slightly to simplify implementation and maybe explanation of the feature. (I will see all 'master' for GitHub remotes that I am not interested higher in the list, but no big deal).
Separate commit, can reverse.